File: 251127/연속되는 수 3/continuous-number3.py
N = int(input())
arr = [int(input()) for _ in range(N)]

count = 0
result = 0
for i in range(N):
    if i >= 1 and (arr[i] < 0 and arr[i - 1] < 0) or (arr[i] > 0 and arr[i - 1] > 0):
        count += 1
    else:
        count = 1
    result = max(result, count)

print(result)

# # 변수 선언 및 입력
# n = int(input())
# arr = [
# 	int(input())
# 	for _ in range(n)
# ]

# # 연속하여 동일한 숫자가 나오는 횟수를 구해보며,
# # 그 중 최댓값을 갱신합니다.
# ans, cnt = 0, 0
# for i in range(n):
# 	# Case 1
# 	if i >= 1 and arr[i] == arr[i - 1]:
# 		cnt += 1
# 	# Case 2
# 	else:
# 		cnt = 1
	
# 	ans = max(ans, cnt)

# print(ans)

File: 260320/연속되는 수 3/continuous-number3.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const arr = input.slice(1, n + 1).map(Number);

// 이전 숫자랑 비교해서 부호가 같으면 (곱해서 양수) ++
let count = 0;
let answer = 0;
for (let i = 0; i < arr.length; i++) {
    if (i >= 1 && arr[i] * arr[i - 1] > 0) {
        count += 1;
    } else {
        count = 1;
    }

    answer = Math.max(count, answer); // 갱신 해줘야 최댓값 저장가능
}
console.log(answer);